library/src/main/java/com/google/android/cameraview/CameraView.java

```diff
@@ -407,13 +411,16 @@ public void takePicture() {
         mImpl.takePicture();
     }
 
-    private class CallbackBridge implements CameraViewImpl.Callback {
+    private static class CallbackBridge implements CameraViewImpl.Callback {
 
         private final ArrayList<Callback> mCallbacks = new ArrayList<>();
 
         private boolean mRequestLayoutOnOpen;
 
-        CallbackBridge() {
+        private WeakReference<CameraView> cameraView;
+
+        CallbackBridge(CameraView cameraView) {
+            this.cameraView = new WeakReference<>(cameraView);
```

Comment:
Since CallbackBridge is now static, we must pass the CameraView.  Within CallbackBridge only hold a weak reference to it.